pkg/aggregator/mocksender/asserts.go

```diff
@@ -14,64 +14,78 @@ import (
+// AssertServiceCheck allows to assert a ServiceCheck was exclusively emitted with given parameters.
 // Additional tags over the ones specified don't make it fail
+// Assert the ServiceCheck wasn't called with any other possible status
 func (m *MockSender) AssertServiceCheck(t *testing.T, checkName string, status metrics.ServiceCheckStatus, hostname string, tags []string, message string) bool {
-	return m.Mock.AssertCalled(t, "ServiceCheck", checkName, status, hostname, AssertTagsContains(t, tags), message)
+	okCall := m.Mock.AssertCalled(t, "ServiceCheck", checkName, status, hostname, AssertTagsContains(tags), message)
+	notOkCalls := m.Mock.AssertNotCalled(t, "ServiceCheck", checkName, AnythingBut(status), hostname, AssertTagsContains(tags), mock.AnythingOfType("string"))
+	return okCall && notOkCalls
 }
 
 // AssertMetric allows to assert a metric was emitted with given parameters.
 // Additional tags over the ones specified don't make it fail
 func (m *MockSender) AssertMetric(t *testing.T, method string, metric string, value float64, hostname string, tags []string) bool {
-	return m.Mock.AssertCalled(t, method, metric, value, hostname, AssertTagsContains(t, tags))
+	return m.Mock.AssertCalled(t, method, metric, value, hostname, AssertTagsContains(tags))
 }
 
 // AssertMetricInRange allows to assert a metric was emitted with given parameters, with a value in a given range.
 // Additional tags over the ones specified don't make it fail
 func (m *MockSender) AssertMetricInRange(t *testing.T, method string, metric string, min float64, max float64, hostname string, tags []string) bool {
-	return m.Mock.AssertCalled(t, method, metric, AssertFloatInRange(t, min, max), hostname, AssertTagsContains(t, tags))
+	return m.Mock.AssertCalled(t, method, metric, AssertFloatInRange(min, max), hostname, AssertTagsContains(tags))
 }
 
 // AssertMetricTaggedWith allows to assert a metric was emitted with given tags, value and hostname not tested.
 // Additional tags over the ones specified don't make it fail
 func (m *MockSender) AssertMetricTaggedWith(t *testing.T, method string, metric string, tags []string) bool {
-	return m.Mock.AssertCalled(t, method, metric, mock.AnythingOfType("float64"), mock.AnythingOfType("string"), AssertTagsContains(t, tags))
+	return m.Mock.AssertCalled(t, method, metric, mock.AnythingOfType("float64"), mock.AnythingOfType("string"), AssertTagsContains(tags))
 }
 
 // AssertMetricNotTaggedWith allows to assert tags were never emitted for a metric.
 func (m *MockSender) AssertMetricNotTaggedWith(t *testing.T, method string, metric string, tags []string) bool {
-	return m.Mock.AssertCalled(t, method, metric, mock.AnythingOfType("float64"), mock.AnythingOfType("string"), AssertTagsNotContains(t, tags))
+	return m.Mock.AssertCalled(t, method, metric, mock.AnythingOfType("float64"), mock.AnythingOfType("string"), AssertTagsNotContains(tags))
+}
+
+// Return a bool value if all the elt of expected are inside the actual array
```

Comment:
I was ready to ask to move it to the public interface, but in fact users should use AssertTagsContains / AssertTagsNotContains

Could you please move it at the very bottom? I like to keep methods sorted from most useful public to least useful public to private, when possible